src/Simplify/AstHelpers.elm

```diff
@@ -752,12 +752,8 @@ expressionContainsAnyVariableInIgnoring bindingsToCheckFor subToIgnore (Node exp
                     expressionContainsAnyVariableInIgnoring bindingsToCheckFor subToIgnore letIn.expression
 
             Expression.CaseExpression caseOf ->
-                -- || TCO-ed
-                if expressionContainsAnyVariableInIgnoring bindingsToCheckFor subToIgnore caseOf.expression then
-                    True
-
-                else
-                    List.any
+                expressionContainsAnyVariableInIgnoring bindingsToCheckFor subToIgnore caseOf.expression
+                    || List.any
```

Comment:
Since there isn't actually optimizable recursion in the `else` branch, we might as well use `||`.

Reply:
I agree, it probably doesn't matter too much 🤷‍♂️